### PR TITLE
Spring Core Module Hometask

### PR DESCRIPTION
**Pull request by brodnikovska**

### CODE REVIEW

File: src/main/java/example/dao/csv/CsvUserDao.java [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    @Override
+    public boolean deleteUser(long id) {
+        return storage.deleteItem(String.format(ID, id));
+    }
+
+    @SneakyThrows
+    @Override
+    public User updateUser(User user) {
+        String newItem = new ObjectMapper().writeValueAsString(user);
+        if (storage.isItemPresent(String.format(ID, user.getId()))) {
```
Comment (andrii-ua-dev) [inline src/main/java/example/dao/csv/CsvUserDao.java]: it is better to move this to private method

String.format(ID, user.getId())

File: src/main/java/example/dao/csv/CsvUserDao.java [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+        }
+        return foundUsers;
+    }
+
+    @SneakyThrows
+    @Override
+    public User createUser(User user) {
+        String newItem = new ObjectMapper().writeValueAsString(user);
+        if (storage.isItemPresent(String.format(ID, user.getId()))) {
+            throw new Exception(String.format("User with id %d is already present", user.getId()));
```
Comment (andrii-ua-dev) [inline src/main/java/example/dao/csv/CsvUserDao.java]: it is better to create your own exception class (e.g. DuplicateException)

File: src/main/java/example/dao/Storage.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+@Component
+@PropertySource(value = "classpath:/application.properties")
+public class Storage {
+
+    @Value("${csv.file.path}")
+    private String csvFilePath;
+
+    private Map<String, String> generalStorage = new HashMap<>();
+
+    @PostConstruct
```
Comment (andrii-ua-dev) [inline src/main/java/example/dao/Storage.java:23]: do we need this annotation? it is configured in xml already.

File: src/main/java/example/model/Event.java [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+import lombok.Getter;
+import lombok.Setter;
+
+import java.util.Date;
+
+@Getter
+@Setter
+public class Event {
+
+    @Column(unique=true)
```
Comment (andrii-ua-dev) [inline src/main/java/example/model/Event.java]: this is a DTO object, Column is redundant here.

File: src/main/java/example/dao/Storage.java [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+
+    public boolean isItemPresent (String id) {
+        return generalStorage.containsKey(id);
+    }
+
+    public boolean isSomeDataPresent (String data) {
+        boolean isPresent = false;
+        List<String> storageValues = generalStorage.values().stream().toList();
+        for (String oneValue : storageValues) {
+            if (oneValue.toLowerCase().contains(data.toLowerCase())) {
```
Comment (andrii-ua-dev) [inline src/main/java/example/dao/Storage.java]: why not to use logic in stream?

File: src/main/java/example/dao/csv/CsvUserDao.java (hunk 2 above)
Comment (brodnikovska) [reply src/main/java/example/dao/csv/CsvUserDao.java]: Done

File: src/main/java/example/dao/Storage.java (hunk 3 above)
Comment (brodnikovska) [reply src/main/java/example/dao/Storage.java:23]: Done

File: src/main/java/example/model/Event.java (hunk 4 above)
Comment (brodnikovska) [reply src/main/java/example/model/Event.java]: Done

File: src/main/java/example/dao/Storage.java (hunk 5 above)
Comment (brodnikovska) [reply src/main/java/example/dao/Storage.java]: Done

**Pull request merged**